virt_lightning/virt_lightning.py

```diff
@@ -197,14 +197,13 @@ def clean_up(self, domain):
         flag |= libvirt.VIR_DOMAIN_UNDEFINE_SNAPSHOTS_METADATA
         domain.dom.undefineFlags(flag)
 
+        self.storage_pool_obj.refresh()
         root = ET.fromstring(xml)
         for disk in root.findall("./devices/disk[@type='file']/source[@file]"):
-            filepath = disk.attrib["file"]
-            try:
-                self.conn.storageVolLookupByPath(filepath).delete()
-            except libvirt.libvirtError as e:
-                if e.get_error_code() == libvirt.VIR_ERR_NO_STORAGE_VOL:
-                    pass
+            filepath = pathlib.PosixPath(disk.attrib["file"])
```

Comment:
Is this save? Does it handle exception if attribute do not exists?

Reply:
Look at the xpath filter above. The attribute will always exists or the filter won't match.